### ISSUE TITLE
While reading a bigquery table, having a JSON type field from Apache Spark throws exception. 

### ISSUE THREAD

**Issue by abhijeet-lele** [GoogleCloudDataproc/spark-bigquery-connector#804]
When Spark tries to read a bigquery table with JSON datatype it throws following exception.

Exception in thread "main" java.lang.IllegalStateException: Unexpected type: JSON
	at com.google.cloud.spark.bigquery.SchemaConverters.getStandardDataType(SchemaConverters.java:327)
	at com.google.cloud.spark.bigquery.SchemaConverters.lambda$getDataType$3(SchemaConverters.java:275)
	at java.util.Optional.orElseGet(Optional.java:267)
	at com.google.cloud.spark.bigquery.SchemaConverters.getDataType(SchemaConverters.java:275)
	at com.google.cloud.spark.bigquery.SchemaConverters.convert(SchemaConverters.java:256)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at com.google.cloud.spark.bigquery.SchemaConverters.toSpark(SchemaConverters.java:59)
	at com.google.cloud.spark.bigquery.BigQueryRelation.schema(BigQueryRelation.java:51)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:449)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:325)
	at org.apache.spark.sql.DataFrameReader.$anonfun$load$3(DataFrameReader.scala:307)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:307)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:225)

**Comment by davidrabinowitz** [comment]
Fixed in version 0.28.0

**Comment by davidrabinowitz** [comment]
Can you please share the table's schema (field names can be obfuscated)?

I cannot reproduce this issue using the latest version.

**Comment by prpretzel** [comment]
Here is the scheme:
![image](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/assets/42389955/30f9b3c6-f001-4af6-844b-79c20105be90)

The version of the connector is not explicitly stated in the code I provided before. I tried to use this:
`.config('spark.jars.packages', 'com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.31.1')`
when I define _spark_ but it didn't change anything. I admit that I could do something wrong and I don't understand what exactly

**Comment by davidrabinowitz** [comment]
Where is your spark installation? Is it possible there is another connector installed? As a way to check that - remove this config and see if you get the same error

**Comment by prpretzel** [comment]
I tried [this](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/issues/804#issuecomment-1584583601) code. There is no config and, as I understand, it should use the latest acceptable version of connector by default
After that I began to specify different versions of connector through `.config()` , but results were the same

**Comment by davidrabinowitz** [comment]
If there is no configuration then it uses the connector installed on the cluster. Where are you running? Dataproc, on prem or another cloud?

**Comment by prpretzel** [comment]
> If there is no configuration then it uses the connector installed on the cluster. Where are you running? Dataproc, on prem or another cloud?

I figured out the problem. It was on my side:
I put an old version of the connector (0.20.0) in `site-packages/pyspark/jars` and totally forget about it. So I replaced it with a new version and now it`s working!
Thanks you, @davidrabinowitz , for your patience!


P.S. I was sure that specifying connector's version using `.config()` has higher priority than locally stored `.jar` file
